### ISSUE TITLE
Question: Muclight vs PubSub for Large Scale Message Delivery

### ISSUE THREAD

**Issue by chrizkiran1** [esl/MongooseIM#4309]
I am evaluating the performance of Muclight and PubSub for handling large-scale message delivery in MongooseIM. Specifically, I need to understand the potential bottlenecks and performance implications of each approach when managing a high number of users.

Scenario:

Muclight Room: 50,000 members 
PubSub Node: 50,000 subscribers

Considering only the message delivery functionality, which approach is likely to cause more bottlenecks in the system? Which one is more efficient for handling such a large number of users?

Any insights or benchmarks comparing the performance of Muclight and PubSub in MongooseIM under these conditions would be greatly appreciated.

**Comment by JanuszJakubiec** [comment]
Hi,

Thank you for reaching out with your question. We've reviewed our past load tests for both `MucLight` and `PubSub` in MongooseIM. Our testing for `MucLight` included scenarios with 30,000 users, where it handled several million messages efficiently. On the other hand, the available `PubSub` tests were conducted with only 3,000 users, so we have limited data on its performance under higher user loads.

Given the available data, I would recommend using `MucLight` for your scenario, as it has demonstrated the capability to manage a user load similar to what you are considering.
